### PR TITLE
feat(standards): require -reusable.yml naming suffix for pure reusables

### PR DESCRIPTION
**Pull request by don-petry**
## What

Follow-up to #626 (merged). Layers a **naming convention** on top of the just-landed "pure reusable workflows must be disabled" standard: a pure reusable (`on: workflow_call` only) MUST be named `*-reusable.yml`.

This was reviewed and approved as part of #626's discussion but landed after that PR auto-merged, so it comes in as its own small PR.

## Why

The `-reusable.yml` suffix makes the "this is a library, safe to disable, never runnable on its own" property legible from the filename — no need to open the file and parse `on:`. The suffix is a **readability aid, not the detection mechanism**: the compliance check still classifies by triggers, never by name (a name can lie — exactly how the `pr-review.yml` engine hid as a pure reusable).

## Changes

- **`standards/ci-standards.md`** — add the naming requirement + rationale to the "Pure reusable workflows must be disabled" section; document the grandfathered `pr-review.yml` exception.
- **`scripts/compliance-audit.sh`** — `check_reusable_workflows_disabled()` now also emits a separate `reusable-naming-<wf>` **warning** for any pure reusable lacking the suffix, independent of the disabled-state finding. `pr-review.yml` is grandfathered until its rename lands.

## Grandfather / follow-up

`pr-review.yml` is the one pure reusable (15/16 already comply) without the suffix; renaming it is entangled with its caller path + 10 `pr-review/*` release-channel tags. Tracked separately in **petry-projects/.github-private#1127**; suppressed in the check until then.

## Validation

- `bash -n` + `shellcheck -S warning` clean.
- Existing bats suite (`reusable-workflows-disabled.bats`, added in #626) still passes — naming block is additive.
- Naming-lint simulated across the live fleet: fires on nothing except the (suppressed) grandfathered `pr-review.yml`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved compliance checks for reusable workflows by adding a naming-suffix validation (pure `workflow_call`-only workflows must end with `-reusable.yml`).
  * The check emits warnings when the suffix is missing, while preserving a grandfathered exception for the legacy PR review workflow.

* **Documentation**
  * Updated CI standards to document the `-reusable.yml` naming requirement and the required manual disablement in the GitHub Actions UI.

* **Tests**
  * Added coverage for compliant and non-compliant filenames, including the exception behavior.

### CODE REVIEW

**Comment by don-petry** [comment]
@coderabbitai resolve

**Comment by don-petry** [comment]
## Dev-Lead — review-changes (applied)
Changes committed and pushed.

File: scripts/compliance-audit.sh [hunk 1]
```diff
@@ -326,6 +326,18 @@ check_reusable_workflows_disabled() {
       continue   # hybrid (workflow_call + real trigger) — exempt, must stay active
     fi
 
+    # Naming lint — a pure reusable must carry the `-reusable.yml` suffix so its
+    # "library, safe to disable, never runnable" nature is legible from the
+    # filename. Detection above is by triggers, never by name (a name can lie);
+    # this is a separate readability finding. Grandfathered: the legacy
+    # `pr-review.yml` engine is tracked for rename in .github-private#1127 —
+    # suppress its naming finding until the rename lands.
+    if [[ "$wf" != *-reusable.yml ]] && [ "$repo/$wf" != ".github-private/pr-review.yml" ]; then
```
[the remark this answers is not in the document]
Comment (don-petry) [reply scripts/compliance-audit.sh]: Fixed in `scripts/compliance-audit.sh` line 335 and the corresponding `_should_flag_naming` helper in `test/scripts/compliance-audit/reusable-workflows-disabled.bats`: merged the two separate bracket tests into a single `[[ "$wf" != *-reusable.yml && "$repo/$wf" != ".github-private/pr-review.yml" ]]` expression, eliminating the mixed `[[`/`[` style.

File: scripts/compliance-audit.sh [hunk 2]
```diff
@@ -326,6 +326,18 @@ check_reusable_workflows_disabled() {
...
+      add_finding "$repo" "reusable-workflows" "reusable-naming-$wf" "warning" \
+        "Pure reusable workflow \`$wf\` (workflow_call-only) does not use the required \`-reusable.yml\` name suffix; rename it to \`<purpose>-reusable.yml\` so its reusable nature is legible from the filename" \
+        "standards/ci-standards.md#pure-reusable-workflows-must-be-disabled"
```
[the remark this answers is not in the document]
Comment (don-petry) [reply scripts/compliance-audit.sh]: The three test cases requested are already present in `test/scripts/compliance-audit/reusable-workflows-disabled.bats` (lines 154–189), added in the same commit as the implementation: (1) `naming: missing suffix is flagged` asserts a pure reusable without the suffix emits a finding, (2) `naming: compliant suffix (-reusable.yml) is NOT flagged` asserts a `*-reusable.yml` file is clean, and (3) `naming: grandfathered pr-review.yml in .github-private is NOT flagged` asserts the exception is suppressed. Two additional edge-case tests cover cross-repo scoping and the `.yaml` (non-`.yml`) extension. The `_should_flag_naming` helper was also updated this session to use consistent `[[ ... ]]` syntax matching the script.

File: standards/ci-standards.md [hunk 3]
```diff
@@ -166,9 +166,21 @@ release-strategy initiative
...
+disable, never runnable on its own" property legible from the filename alone — a
+reader (or an auditor) should not have to open the file and parse its `on:` block
+to know it is a reusable. The suffix is a **readability aid, not the detection
+mechanism**: compliance tooling classifies by triggers, never by name (a name can
+lie — see the `pr-review.yml` engine, a pure reusable that predates this rule).
+When you extract logic into a new reusable, name it `<purpose>-reusable.yml` from
+the start. *Grandfathered exception:* `.github-private/.github/workflows/pr-review.yml`
+is a pure reusable that keeps its legacy name until it can be renamed in lockstep
+with its caller path and `pr-review/*` release-channel tags — tracked in
+`petry-projects/.github-private#1127`.
```
[the remark this answers is not in the document]
Comment (don-petry) [reply standards/ci-standards.md:183]: Updated the **Compliance** paragraph in `standards/ci-standards.md` (after line 227) to add a second sentence describing the new naming-suffix check: it now states that the function also emits a separate `warning` finding for any pure reusable whose filename lacks the `-reusable.yml` suffix, with the grandfathered `pr-review.yml` exception noted and linked to the tracking issue.

**Comment by don-petry** [comment]
## Dev-Lead — review-changes (applied)
Changes committed and pushed.

**Comment by don-petry** [comment]
## Dev-Lead — fix-bot-comment (no-changes)
<details><summary>Agent reasoning</summary>

~~~~
Issues addressed: 0
Files changed: None required
Skipped (informational): 0
**Status:** ✅ No actionable issues. SonarCloud Quality Gate passed with no new issues, no security hotspots, and all CI checks passing or skipped. The PR is ready to merge from a quality/security perspective.
~~~~
</details>

**Comment by don-petry** [comment]
## Dev-Lead — fix-bot-comment (no-changes)
<details><summary>Agent reasoning</summary>

~~~~
- SonarCloud quality gate: **PASSED** (all checks green)
- Review threads: **All resolved** (confirmed via GraphQL query)
- Prior CodeRabbit stale-doc finding: **Already addressed** — standards/ci-standards.md lines 222–231 now document both the pure reusable disabled-state check and the separate `-reusable.yml` naming check, matching the implementation in scripts/compliance-audit.sh lines 329–339
- No Tier 1 blockers (no failures, CHANGES_REQUESTED, or other blocking states)
**Files verified:**
- `standards/ci-standards.md` — compliance section updated with reusable naming requirement and grandfathered exception documentation
- `scripts/compliance-audit.sh` — reusable naming lint implemented with proper exemption for legacy `pr-review.yml`
- `test/scripts/compliance-audit/reusable-workflows-disabled.bats` — 5 new tests added covering the naming check
**Issues addressed:** 0 (quality gate passed with no findings)
No changes needed — the PR is ready.
~~~~
</details>
